subsys/cpool/src/main/java/org/commonjava/indy/subsys/cpool/ConnectionPoolConfig.java

```diff
@@ -0,0 +1,109 @@
+package org.commonjava.indy.subsys.cpool;
+
+import org.commonjava.indy.conf.IndyConfigInfo;
+import org.commonjava.web.config.ConfigurationException;
+import org.commonjava.web.config.annotation.SectionName;
+import org.commonjava.web.config.section.MapSectionListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.lang.Boolean.TRUE;
+import static org.commonjava.indy.subsys.cpool.ConnectionPoolConfig.SECTION;
+
+@ApplicationScoped
+@SectionName(SECTION)
+public class ConnectionPoolConfig
+        extends MapSectionListener
+        implements IndyConfigInfo
+{
```

Comment:
One problem here. Our db info can not be stored in a config file in upshift env. They are all stored in secrets (/mnt/secrets). I'm not sure if this config has enabled this part of support, so just a mark here.

Reply:
We should develop a feature to read secrets files, but it looks compatible with what we have now.